src/wasm/wasm-type.cpp

```diff
@@ -1781,6 +1825,12 @@ std::ostream& TypePrinter::print(HeapType type) {
     case HeapTypeKind::Basic:
       WASM_UNREACHABLE("unexpected kind");
   }
+  if (type.getDescriptorType()) {
+    os << ')';
+  }
+  if (type.getDescribedType()) {
+    os << ')';
+  }
```

Comment:
Huh, I see that the current overview does allow a thing to be both a descriptor and describe...